### PR TITLE
schema,server,api: events improvement

### PR DESCRIPTION
**Pull request by shwstppr**
### Description

Add resource ID and resource type to event.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### Screenshots (if appropriate):
Screenshot from 2022-03-24 14-21-27

### CODE REVIEW

**Comment by nvazquez** [comment]
Hi @shwstppr can you please fix the conflicts?

**Comment by shwstppr** [comment]
@blueorangutan package

File: server/src/main/java/org/apache/cloudstack/outofbandmanagement/OutOfBandManagementServiceImpl.java [hunk 1]
```diff
@@ -16,6 +16,36 @@
...
+import org.apache.cloudstack.outofbandmanagement.dao.OutOfBandManagementDao;
+import org.apache.cloudstack.outofbandmanagement.driver.OutOfBandManagementDriverChangePasswordCommand;
+import org.apache.cloudstack.outofbandmanagement.driver.OutOfBandManagementDriverPowerCommand;
+import org.apache.cloudstack.outofbandmanagement.driver.OutOfBandManagementDriverResponse;
+import org.apache.cloudstack.poll.BackgroundPollManager;
+import org.apache.cloudstack.poll.BackgroundPollTask;
+import org.apache.cloudstack.utils.identity.ManagementServerNode;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.log4j.Logger;
+import org.springframework.stereotype.Component;
```
Comment (GutoVeronezi) [inline server/src/main/java/org/apache/cloudstack/outofbandmanagement/OutOfBandManagementServiceImpl.java:47]: Is this reordering necessary?

File: server/src/main/java/com/cloud/storage/snapshot/SnapshotManagerImpl.java [hunk 2]
```diff
@@ -60,7 +61,10 @@
 import org.apache.cloudstack.storage.datastore.db.SnapshotDataStoreDao;
 import org.apache.cloudstack.storage.datastore.db.SnapshotDataStoreVO;
 import org.apache.cloudstack.storage.datastore.db.StoragePoolVO;
+import org.apache.cloudstack.utils.reflectiontostringbuilderutils.ReflectionToStringBuilderUtils;
 import org.apache.commons.collections.MapUtils;
+import org.apache.commons.lang3.builder.ReflectionToStringBuilder;
+import org.apache.commons.lang3.builder.ToStringStyle;
```
Comment (GutoVeronezi) [inline server/src/main/java/com/cloud/storage/snapshot/SnapshotManagerImpl.java:67]: Is this reordering necessary?

File: server/src/main/java/com/cloud/resource/RollingMaintenanceManagerImpl.java [hunk 3]
```diff
@@ -16,6 +16,29 @@
...
+
+import org.apache.cloudstack.affinity.AffinityGroupProcessor;
+import org.apache.cloudstack.api.ApiCommandJobType;
+import org.apache.cloudstack.api.command.admin.cluster.UpdateClusterCmd;
+import org.apache.cloudstack.api.command.admin.host.PrepareForMaintenanceCmd;
+import org.apache.cloudstack.api.command.admin.resource.StartRollingMaintenanceCmd;
+import org.apache.cloudstack.context.CallContext;
+import org.apache.cloudstack.framework.config.ConfigKey;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.log4j.Logger;
```
Comment (GutoVeronezi) [inline server/src/main/java/com/cloud/resource/RollingMaintenanceManagerImpl.java:40]: Is this reordering necessary?

File: plugins/network-elements/ovs/src/main/java/com/cloud/network/guru/OvsGuestNetworkGuru.java [hunk 4]
```diff
@@ -17,6 +17,12 @@
 package com.cloud.network.guru;
 
 
+import javax.inject.Inject;
+
+import org.apache.cloudstack.context.CallContext;
+import org.apache.log4j.Logger;
+import org.springframework.stereotype.Component;
```
[the remark this answers is not in the document]
Comment (shwstppr) [reply plugins/network-elements/ovs/src/main/java/com/cloud/network/guru/OvsGuestNetworkGuru.java:24]: Keeps them grouped and alphabetically ordered, improves readability

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan test

**Comment by shwstppr** [comment]
@blueorangutan test

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan test

**Comment by shwstppr** [comment]
@blueorangutan test

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan test

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan test

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan test

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan test keepEnv

**Comment by shwstppr** [comment]
@blueorangutan test keepEnv

**Comment by shwstppr** [comment]
@blueorangutan test keepEnv

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan test

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan test keepEnv

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan test

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan test

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan test

**Comment by shwstppr** [comment]
@blueorangutan test

**Review by harikrishna-patnala — APPROVED** [review]
LGTM, tested with a few operations like VM deploy, volume creation, snapshots, events section shows the resource involved and also the type when hovering on the resource icon.

**Comment by nvazquez** [comment]
Hi @shwstppr can you please fix the conflicts?

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan test

**Comment by shwstppr** [comment]
@blueorangutan test

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Comment by nvazquez** [comment]
@shwstppr sorry it seems there are conflicts again, can you fix?

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by rohityadavcloud** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan test

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by vladimirpetrov** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by vladimirpetrov** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by vladimirpetrov** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@shwstppr can you please fix the conflicts?

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan test

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan test

**Review by rohityadavcloud — APPROVED** [review]

**Comment by rohityadavcloud** [comment]
@shwstppr can you check and rekick Travis failure (some issue with db upgrade path)?

**Review by nvazquez — APPROVED** [review]

**Pull request merged**